### PR TITLE
DAOS-4474 test: Reset pool handle and cont UUID when open fails

### PR DESCRIPTION
**Pull request by shimizukko**
When we use invalid pool handle or UUID to open a container
and the container open fails, it causes an error during
tearDown. To avoid the error, reset them to valid values after
open fails.

Test-tag: pr,-hw container_open
Signed-off-by: Makito Kano <makito.kano@intel.com>

### CODE REVIEW

**Comment by shimizukko** [comment]
Test Failures:
Functional / 18-./daos_test/daos_core_test.py:DaosCoreTest.test_subtest;test_z-num_clients-num_replicas-hosts-server_config-5885 – DaosCoreTest -> DAOS-3945
Functional / 15-./pool/info_tests.py:InfoTests.test_pool_info_query;hosts-mode_rw-createset-size1gb-servers-ex-d5f0 – InfoTests -> DAOS-3945
Functional_Hardware_Large / 1-./io/mdtest_small.py:MdtestSmall.test_mdtest_small;container-dfuse-hosts-np_8-pool-server_config-cbc2 – MdtestSmall -> DAOS-4571

File: src/tests/ftest/container/open.py [hunk 1]
```diff
@@ -125,24 +123,22 @@ def test_container_open(self):
...
             self.container[-1].get_params(self)
             self.container[-1].create()
             container_uuids.append(uuid.UUID(self.container[-1].uuid))
-            i += 1
 
         # Decide which pool handle and container UUID to use. The PASS/FAIL
         # number corresponds to the index for self.pool and container_uuids
         pool_handle_index = poh_state_num
         container_uuid_index = uuid_state_num
+        open_failed = False
```
Comment (phender) [inline src/tests/ftest/container/open.py:141]: There is a `TestContainer.opened` (https://github.com/daos-stack/daos/blob/master/src/tests/ftest/util/test_utils_container.py#L394) property that should negate the need for this test to maintain its own flag.  This value is read during `TestWithServers.destroy_containers`: https://github.com/daos-stack/daos/blob/master/src/tests/ftest/util/apricot/apricot/test.py#L622

**Comment by shimizukko** [comment]
Failed Tests:
Functional / 1-./pool/connect_test.py:ConnectTest.test_connect;hosts-server_config-validsetname-2abf – ConnectTest -> DAOS-4687
Functional / 1-./pool/multiple_creates_test.py:MultipleCreatesTest.test_create_one;hosts-server_config-modeall-validsetname-9bc0 – MultipleCreatesTest -> DAOS-4687
Functional / 2-./pool/multiple_creates_test.py:MultipleCreatesTest.test_create_two;hosts-server_config-modeall-validsetname-9bc0 – MultipleCreatesTest -> DAOS-4687
Functional / 3-./pool/multiple_creates_test.py:MultipleCreatesTest.test_create_three;hosts-server_config-modeall-validsetname-9bc0 – MultipleCreatesTest -> DAOS-4687
Functional_Hardware_Large / 1-./io/mdtest_small.py:MdtestSmall.test_mdtest_small;container-dfuse-hosts-np_8-pool-server_config-694e – MdtestSmall -> DAOS-4285
Functional_Hardware_Medium / 10-./daos_test/daos_core_test-rebuild.py:DaosCoreTestRebuild.test_rebuild;test_r_26-num_clients-num_replicas-hosts-0-1-f547 – DaosCoreTestRebuild -> DAOS-4302

File: src/tests/ftest/container/open.py (hunk 1 above)
Comment (shimizukko) [reply src/tests/ftest/container/open.py]: I use the opened value to check if open failed and it works. Thanks for the suggestion.

File: src/tests/ftest/container/open.py [hunk 2]
```diff
@@ -170,3 +166,14 @@ def test_container_open(self):
...
+        # Calling container.open with invalid pool handle or UUID results in
+        # setting them in the underlying DaosContainer's member. When we try to
+        # destroy it with API in tearDown, it'll pass in the invalid
+        # handle/UUID member to the API and causes an error. Set
+        # self.container = None so that container destroy doesn't occur during
+        # tearDown. Note that we shouldn't set None when open succeeds because
+        # in that case, the container needs to be properly destroyed for the
+        # pool destroy process to work during tearDown.
+        if not self.container[0].opened or not self.container[1].opened:
+            self.container = None
```
Comment (phender) [inline src/tests/ftest/container/open.py:179]: Shouldn't we just reset the handle/UUID to the correct value(s) so that `tearDown()` works correctly?

**Comment by shimizukko** [comment]
Test Failures:
Functional_Hardware_Large / 1-./io/mdtest_small.py:MdtestSmall.test_mdtest_small;container-dfuse-hosts-np_8-pool-server_config-694e – MdtestSmall -> DAOS-4285
Functional_Hardware_Medium / 04-./daos_test/daos_core_test-rebuild.py:DaosCoreTestRebuild.test_rebuild;test_r_20-num_clients-num_replicas-hosts-0-1-b892 – DaosCoreTestRebuild -> DAOS-3945
Functional_Hardware_Medium / 06-./daos_test/daos_core_test-rebuild.py:DaosCoreTestRebuild.test_rebuild;test_r_22-num_clients-num_replicas-hosts-0-1-24a6 – DaosCoreTestRebuild -> DAOS-3952
Functional_Hardware_Medium / ALL – rebuild_tests_22 -> DAOS-3952

File: src/tests/ftest/container/open.py (hunk 2 above)
Comment (shimizukko) [reply src/tests/ftest/container/open.py]: Done

**Pull request merged**